### PR TITLE
feat: archivist & attestation source integration

### PR DESCRIPTION
**Pull request by mikhailswift**
This uses the updated attestation source API from go-witness during
policy verification. This also adds support for storage of attestations
within archivist and using archivist to verify policies.

Signed-off-by: Mikhail Swift <mikhail@testifysec.com>

### CODE REVIEW

**Comment by mikhailswift** [comment]
This is a draft but all existing tests are working.

Archivist functionality needs to be thoroughly tested.

**Comment by mikhailswift** [comment]
This requires https://github.com/testifysec/go-witness/pull/10

The go.mod has been updated to point at a commit on the go-witness branch but a go-witness release will need to be cut and this project's go.mod updated once ready.

**Comment by mikhailswift** [comment]
Docs also need to be updated

File: cmd/run_test.go [hunk 1]
```diff
@@ -16,34 +16,36 @@ package cmd
...
 	"testing"
 
 	"github.com/testifysec/go-witness/cryptoutil"
+	"github.com/testifysec/go-witness/dsse"
 	"github.com/testifysec/witness/options"
 )
 
-func Test_runRunRSAKeyPair(t *testing.T) {
-
+func TestRunRSAKeyPair(t *testing.T) {
```
Comment (mikhailswift) [inline cmd/run_test.go:35]: Were these tests generated by copilot?

They seem odd

File: cmd/verify.go [hunk 2]
```diff
@@ -87,45 +86,40 @@ func runVerify(vo options.VerifyOptions, args []string) error {
...
-		}
-
-		digestSets := []cryptoutil.DigestSet{}
-		digestSets = append(digestSets, artifactDigestSet)
-
-		verifiers := []cryptoutil.Verifier{}
-		verifiers = append(verifiers, verifier)
+	subjects := []cryptoutil.DigestSet{artifactDigestSet}
+	for _, subDigest := range vo.AdditionalSubjects {
+		subjects = append(subjects, cryptoutil.DigestSet{crypto.SHA256: subDigest})
```
Comment (mikhailswift) [inline cmd/verify.go:96]: This is less than ideal.

We probably need to be parsing `AdditionalSubjects` and making sure they're valid digests prior to using them

**Comment by mikhailswift** [comment]
Will require https://github.com/testifysec/go-witness/pull/21 and a go.mod bump before merging

**Review by colek42 — DISMISSED** [review]

**Review by colek42 — APPROVED** [review]

**Pull request merged**